### PR TITLE
Remove ToS acceptance copy from all entrypoints in MOX

### LINKED ISSUE #8167: Remove ToS acceptance copy from all entry points into MOX
### Description

Since we now include the WooPayments ToS acceptance (and Privacy Policy) in the first step of the MOX, we should remove the ToS copy from all other places to avoid duplicative copy and bad UX (the ToS copy adds friction and design challenges especially in "cramped UI" conditions.).

### Acceptance criteria

WooPayments ToS copy is removed from:
* WooCommerce task list (with or without active incentive - if there are incentive-specific T&C, those should be kept)
* Payments > Connect page (with or without active incentive - if there are incentive-specific T&C, those should be kept)
* WooCommerce > Settings > Payments promo banner and payment methods list (see tvclrJLW8LJHhOojcKZGy6-fi-4_1074 )
* If there are other places that you can find, remove from those places also.

cc @elizaan36

### PR DESCRIPTION
**Pull request by oaratovskyi**
Fixes #8167 

#### Changes proposed in this Pull Request

Removed ToS acceptance from:
- Settings -> Payment methods

Removing ToS acceptance from Core is done in https://github.com/woocommerce/woocommerce/pull/46003

p.s. Removal from the connect page was done earlier, same for WC Core task list

#### Testing instructions

* Checkout the branch `add/8167-remove-tos-links-that-are-not-needed-anymore`
* Visit settings page and check that no ToS acceptance part is present anymore.

#### Screenshots
| Before | After |
|---|---|
| image | image |

-------------------

- [x] Run `npm run changelog` to add a changelog file, choose `patch` to leave it empty if the change is not significant. You can add multiple changelog files in one PR by running this command a few times. 
- [x] Covered with tests (or have a good reason not to test in description ☝️)
- [x] Tested on mobile (or does not apply)

**Post merge**

- [x] Link to testing instructions from [release testing doc](https://github.com/Automattic/woocommerce-payments/wiki/Release-testing-instructions) following [these instructions](https://github.com/Automattic/woocommerce-payments/wiki/How-to-write-good-manual-testing-scenarios) : _'QA Testing Not Applicable'_
- [x] Add or update [critical flows](https://github.com/Automattic/woocommerce-payments/wiki/Critical-flows) and [testing instructions for critical flows](https://github.com/Automattic/woocommerce-payments/wiki/Testing-instructions-for-critical-flows), if applicable.
- [x] Add what's changed (description, screenshot, demo videos etc.) to the release announcement post, if applicable.

### CODE REVIEW

**Comment by botwoo** [comment]
### Test the build 

#### Option 1. Jetpack Beta

- Install and activate [Jetpack Beta](https://jetpack.com/download-jetpack-beta/).
- Use this build by searching for PR number `8450` or branch name `add/8167-remove-tos-links-that-are-not-needed-anymore` in your-test.site/wp-admin/admin.php?page=jetpack-beta&plugin=woocommerce-payments          

#### Option 2. Jurassic Ninja - available for logged-in A12s

:rocket: [Launch a JN site with this branch](https://jurassic.ninja/create/?jetpack-beta&shortlived&nojetpack&woocommerce&woocommerce-payments-dev-tools&branches.woocommerce-payments=add/8167-remove-tos-links-that-are-not-needed-anymore) :rocket:

:information_source: Install this [Tampermonkey script](https://github.com/Automattic/woocommerce-payments/tree/develop/bin/wcpay-live-branches) to get more options.

---

Build info: 

- Latest commit: 5b2ffc625dc0dc97ac5e19ceecebfa82fd588299
- Build time: 2024-03-27 15:30:19 UTC

_Note: the build is updated when a new commit is pushed to this PR._

File: client/connect-account-page/strings.tsx [hunk 1]
```diff
@@ -38,62 +38,6 @@ export default {
 		'Earn recurring revenue and get deposits into your bank account.',
 		'woocommerce-payments'
 	),
-	agreement: createInterpolateElement(
```
Comment (oaratovskyi) [inline client/connect-account-page/strings.tsx:41]: from connect page, not used anymore

**Review by mordeth — APPROVED** [review]
✅ Code changes are looking good!
✅ Manual tests are successful.

:shipit: LGTM!

**Review by elazzabi — APPROVED** [review]
Looks good 🎉

**Pull request merged**